map-low.go

```diff
@@ -416,22 +423,23 @@ func (m *BPFMapLow) UpdateBatch(keys, values unsafe.Pointer, count uint32) error
 		optsC,
 	)
 	errno = syscall.Errno(-retC)
-	if retC < 0 {
-		if errno != syscall.EFAULT && uint32(countC) != count {
-			return fmt.Errorf("failed to update ALL elements in map %s, updated (%d/%d): %w", m.Name(), uint32(countC), count, errno)
-		}
-		return fmt.Errorf("failed to batch update elements in map %s: %w", m.Name(), errno)
+	// retC < 0 && errno == syscall.E2BIG indicates a partial update.
+	if retC < 0 && (errno != syscall.E2BIG || countC == 0) {
```

Comment:
Ohh good catch, I think I introduced this bug. I was trying to find where in the kernel E2BIG is returned if there's a partial update and couldn't find that code path. Would you mind sharing it? Thanks!!

Reply:
I've followed this path:

https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L5401
https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L4883
https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L4845
https://elixir.bootlin.com/linux/latest/source/kernel/bpf/arraymap.c#L764
https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L1750
https://elixir.bootlin.com/linux/latest/source/kernel/bpf/syscall.c#L183
https://elixir.bootlin.com/linux/latest/source/kernel/bpf/arraymap.c#L853

But the initial hint was the return types from sys_bpf: https://man7.org/linux/man-pages/man2/bpf.2.html#ERRORS

```
ERRORS
       E2BIG  The eBPF program is too large or a map reached the max_entries limit (maximum
              number of elements).
```

Reply:
Thanks for the context, I hadn't checked array maps where perhaps this is more obvious. Should we then augment the tests to check that the partial / total batch operations work well with other map types such as arrays?